sparse/impl/KokkosSparse_sptrsv_cuSPARSE_impl.hpp

```diff
@@ -60,6 +60,142 @@ void sptrsvcuSPARSE_symbolic(KernelHandle* sptrsv_handle,
+    } else if (std::is_same<scalar_type, Kokkos::complex<float>>::value) {
+      cudaValueType = CUDA_C_32F;
+    } else {
+      throw std::runtime_error("CUSPARSE wrapper error: unsupported type.\n");
+    }
+
+    // Create sparse matrix in CSR format
+    status = cusparseCreateCsr(&(h->matDescr), static_cast<int64_t>(nrows),
+                               static_cast<int64_t>(nrows), nnz, rm,
+                               (void*)entries.data(), (void*)values.data(),
+                               cudaCsrRowMapType, cudaCsrColIndType,
+                               CUSPARSE_INDEX_BASE_ZERO, cudaValueType);
+
+    // Create dummy dense vector B (RHS)
+    nnz_scalar_view_t b_dummy("b_dummy", nrows);
+    cusparseCreateDnVec(&(h->vecBDescr_dummy), static_cast<int64_t>(nrows),
+                        b_dummy.data(), cudaValueType);
+
+    // Create dummy dense vector X (LHS)
+    nnz_scalar_view_t x_dummy("x_dummy", nrows);
+    cusparseCreateDnVec(&(h->vecXDescr_dummy), static_cast<int64_t>(nrows),
+                        x_dummy.data(), cudaValueType);
+
+    // Specify Lower|Upper fill mode
+    if (is_lower) {
+      cusparseFillMode_t fillmode = CUSPARSE_FILL_MODE_LOWER;
+      cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_FILL_MODE,
+                                &fillmode, sizeof(fillmode));
+    } else {
+      cusparseFillMode_t fillmode = CUSPARSE_FILL_MODE_UPPER;
+      cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_FILL_MODE,
+                                &fillmode, sizeof(fillmode));
+    }
+
+    // Specify Unit|Non-Unit diagonal type.
+    cusparseDiagType_t diagtype = CUSPARSE_DIAG_TYPE_NON_UNIT;
+    cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_DIAG_TYPE, &diagtype,
+                              sizeof(diagtype));
+
+    // Allocate an external buffer for analysis
+    status = cusparseSpSV_bufferSize(
+        h->handle, h->transpose, &alpha, h->matDescr, h->vecBDescr_dummy,
+        h->vecXDescr_dummy, cudaValueType, CUSPARSE_SPSV_ALG_DEFAULT,
+        h->spsvDescr, &pBufferSize);
+
+    if (CUSPARSE_STATUS_SUCCESS != status)
+      std::cout << "bufferSize status error name " << (status) << std::endl;
+
+    // pBuffer returned by cudaMalloc is automatically aligned to 128 bytes.
+    cudaError_t my_error;
+    my_error = cudaMalloc((void**)&(h->pBuffer), pBufferSize);
+
+    if (cudaSuccess != my_error)
+      std::cout << "cudmalloc pBuffer error_t error name "
+                << cudaGetErrorString(my_error) << std::endl;
+
+    // Run analysis
+    status = cusparseSpSV_analysis(h->handle, h->transpose, &alpha, h->matDescr,
+                                   h->vecBDescr_dummy, h->vecXDescr_dummy,
+                                   cudaValueType, CUSPARSE_SPSV_ALG_DEFAULT,
+                                   h->spsvDescr, h->pBuffer);
+
+    if (CUSPARSE_STATUS_SUCCESS != status)
+      std::cout << "analysis status error name " << (status) << std::endl;
+  } else {
```

Comment:
Small nit-pick, verifying all the required conditions at the top (or bottom) of the if-statement would make it easier to maintain in my opinion
```
if(!is_cuda) { throw ... }
else if(!is_valid_ordinal) { throw... }
else if(!is_another_condition) { throw...}
else {

// the actual implementation

}
```

Reply:
@lucbv Changes were made.